engine.py

```diff
@@ -912,3 +912,58 @@ def __setup_favorite_dirs(self):
                                 type=(nuke.IMAGE | nuke.SCRIPT | nuke.GEO),
                                 icon=icon_path,
                                 tooltip=path)
+
+    #####################################################################################
+    # Script and session related methods
+
+    def get_session_path(self):
+        """
+        Returns the path to the current file if it resides on
+        disk. If unsaved, it returns None.
+
+        :return: Path to the current scene if file is saved, else returns
+                 and empty string.
+        """
+        # retrieve the name of the nuke file. In case of a saved file, the
+        # absolute path of the file is returned, for an unsaved file,
+        # an empty string is returned.
+        return nuke.root().knob("name").value()
+
+    def get_session_dependencies(self):
```

Comment:
This does not reflect the core interface - a `session=None` parameter is missing!